test_conformance/images/kernel_read_write/test_cl_ext_image_from_buffer.cpp

```diff
@@ -0,0 +1,873 @@
+
+            int fill_error = image_from_buffer_fill_check(
+                queue, image_from_buffer, region, element_size, pattern);
+            if (TEST_PASS != fill_error)
+            {
+                return fill_error;
+            }
+
+            fill_error = image_from_buffer_fill_check(queue, image, region,
+                                                      element_size, pattern);
+            if (TEST_PASS != fill_error)
+            {
+                return fill_error;
+            }
+
+            err = clReleaseMemObject(buffer);
+            test_error(err, "Unable to release buffer");
+
+            err = clReleaseMemObject(image);
+            test_error(err, "Unable to release image");
+
+            err = clReleaseMemObject(image_from_buffer);
+            test_error(err, "Unable to release image");
+        }
+    }
+
+    return TEST_PASS;
+}
+
+static int image_from_buffer_read_check(cl_command_queue queue, cl_mem buffer,
+                                        const size_t buffer_size,
+                                        size_t* region, size_t element_size,
+                                        char pattern, size_t row_pitch,
+                                        size_t slice_pitch)
+{
+    /* read the buffer and check the pattern */
+    std::vector<char> host_buffer(buffer_size);
+    char* host_ptr = host_buffer.data();
+    char* host_ptr_slice = host_ptr;
+
+    cl_int error =
+        clEnqueueReadBuffer(queue, buffer, CL_BLOCKING, 0, buffer_size,
+                            host_buffer.data(), 0, nullptr, nullptr);
+    test_error(error, "Error clEnqueueReadBuffer");
+
+    for (size_t k = 0; k < region[2]; k++)
+    {
+        for (size_t i = 0; i < region[1]; i++)
+        {
+            for (size_t j = 0; j < region[0] * element_size; j++)
+            {
+                if (host_ptr[j] != pattern)
+                {
+                    test_fail("Image pattern check failed");
+                }
+            }
+            host_ptr = host_ptr + row_pitch;
+        }
+        host_ptr_slice = host_ptr_slice + slice_pitch;
+        host_ptr = host_ptr_slice;
+    }
+
+    return TEST_PASS;
+}
+
+/**
+ * Use fill image to fill the buffer that was used to create the image
+ */
+int image_from_buffer_read_positive(cl_device_id device, cl_context context,
+                                    cl_command_queue queue)
+{
+    if (!is_extension_available(device, "cl_ext_image_requirements_info"))
+    {
+        printf("Extension cl_ext_image_requirements_info not available");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    if (!is_extension_available(device, "cl_ext_image_from_buffer"))
+    {
+        printf("Extension cl_ext_image_from_buffer not available");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    std::vector<cl_mem_object_type> imageTypes{
+        CL_MEM_OBJECT_IMAGE1D,       CL_MEM_OBJECT_IMAGE2D,
+        CL_MEM_OBJECT_IMAGE3D,       CL_MEM_OBJECT_IMAGE1D_BUFFER,
```

Comment:
This test does not supply a mem_object/buffer in the image descriptor, which we believe is invalid for `CL_MEM_OBJECT_IMAGE1D_BUFFER` types.

The way in which we have interpreted the wording in `cl_ext_image_from_buffer` still requires that a buffer object is supplied when creating a `CL_MEM_OBJECT_IMAGE1D_BUFFER` object.